utils/record/record.go

```diff
@@ -279,10 +279,19 @@ func (e *EventRecorderAdapter) sendNotifications(object runtime.Object, opts Eve
 	}
 
 	for _, dest := range destinations {
-		err = notificationsAPI.Send(objMap, triggerActions[0].Send, dest)
+		res, err := notificationsAPI.RunTrigger(trigger, objMap)
 		if err != nil {
-			log.Errorf("notification error: %s", err.Error())
-			return err
+			log.Errorf("Failed to execute condition of trigger %s: %v", trigger, err)
```

Comment:
There is probably a missing `continue` or `return err` statement below this line otherwise it will just log and continue executing in the current iteration.

Reply:
done